music_assistant/providers/hass/__init__.py

```diff
@@ -445,16 +445,19 @@ async def _handle_player_control_mute_set(self, entity_id: str, muted: bool) ->
 
     async def _handle_player_control_volume_set(self, entity_id: str, volume_level: int) -> None:
         """Handle setting volume on the playercontrol."""
-        if entity_id.startswith("media_player."):
+        domain = entity_id.split(".", 1)[0]
+
+        if domain == "media_player":
             await self.hass.call_service(
-                domain="media_player",
+                domain=domain,
                 service="volume_set",
                 service_data={"volume_level": volume_level / 100},
                 target={"entity_id": entity_id},
             )
         else:
```

Comment:
let's use a guard clause here. return from the media_player block and

Reply:
You can prevent the else clause in a short function like this by returning early.
so you handle the top check and return afterwards. Its considered more pythonic and makes the code friendlier to read

Reply:
I was on the mindset of "minimizing changes to original codebase" and it didn't cross my mind.

Makes sense, fixed in the following commit.